### PR TITLE
docs: add protocol section

### PR DESCRIPTION
**Pull request by KaiserKarel**

### CODE REVIEW

File: .github/workflows/deploy-docs.yml [hunk 1]
```diff
@@ -63,7 +63,6 @@ jobs:
 
   deploy-preview:
     runs-on: ['ubuntu-latest']
-    needs: [garnix]
```
Comment (PoisonPhang) [inline .github/workflows/deploy-docs.yml:66]: This will result in the package being built on the GH runner

Comment (PoisonPhang) [reply .github/workflows/deploy-docs.yml:66]: Also, if we're committing to that, we should do it for all deployments

Comment (KaiserKarel) [reply .github/workflows/deploy-docs.yml:66]: See the chat. We want deployment URLs asap, even if other checks still fail.

**Review by PoisonPhang — APPROVED** [review]

File: docs/src/content/docs/protocol/overview.mdx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+Union supports different protocols over each channel. Channels are connection agnostic, meaning that all channel versions support both direct and recursive connections.
+
+<Mermaid
+  content={`
+flowchart TB
+    Connection --> Channel
+    Channel --> ICS20-1-relay
+    Channel --> UCS0-1-relay
+    Channel --> UCS0-2-relay
+    Channel --> UCS0-3-relay
```
Comment (benluelo) [inline docs/src/content/docs/protocol/overview.mdx:50]:
```suggestion
    Channel --> ics20-1
    Channel --> ucs01-relay
    Channel --> ucs02-nft
    Channel --> ucs03-???
```

File: docs/src/content/docs/protocol/channels/ucs02-relay.mdx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+---
+title: "ucs02-relay-1"
```
Comment (benluelo) [inline docs/src/content/docs/protocol/channels/ucs02-relay.mdx:2]: ucs02-nft-1

File: docs/src/content/docs/protocol/channels/ucs03-relay.mdx [hunk 4]
```diff
@@ -0,0 +1,90 @@
+---
+title: "ucs03-relay-1"
```
Comment (benluelo) [inline docs/src/content/docs/protocol/channels/ucs03-relay.mdx:2]: we haven't officially named this protocol yet

File: docs/src/content/docs/protocol/channels/ucs02-nft.mdx [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    variant: caution
+---
+
+# Overview
+
+`ucs02-nft-1` is the first IBC NFT transfer protocol with EVM capabilities. The functionality has been moved to `ucs03-zkgm-1`.
+
+:::caution
+
+`ucs02-nft-1` is deprecated. Migrate to [`ucs03-zkgm-1`](/protocol/channels/ucs03-relay).
```
Comment (benluelo) [inline docs/src/content/docs/protocol/channels/ucs02-nft.mdx]: link

**Review by hussein-aitlahcen — DISMISSED** [review]

File: docs/src/content/docs/protocol/connections/direct.mdx [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+Direct IBC connections are what most chains leverage, and are currently the only option when leveraging the vanilla IBC stack. Union's hub-and-spoke model is
+supported, but has to leverage [packet-forward middleware](https://github.com/cosmos/ibc-apps/tree/main/middleware/packet-forward-middleware). There are a few
+fundamental downsides to PFM, described in our research on 
+[statelenses](https://research.union.build/Union-Research-4f780b07025b4c9b8014b8aeea68dbad?p=9e3d6578ec0e48fca8e502a0d28f485c&pm=s) 
+and [intents](https://research.union.build/Union-Research-4f780b07025b4c9b8014b8aeea68dbad?p=10c79e5af45d80a3a0f8dde7159138af&pm=s).
+
+:::tip
+
+Use [`recursive connections`](/protocol/connections/direct) and [`ucs03-zkgm-1`](/protocol/channels/ucs03-relay) instead for faster bridging and lower costs.
```
Comment (benluelo) [inline docs/src/content/docs/protocol/connections/direct.mdx]: link

File: docs/src/content/docs/protocol/overview.mdx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  order: 1
+---
+
+import Mermaid from "#/components/Mermaid.astro";
+
+The Union protocol is a flavor of IBC (and fully IBC compatible), with certain extensions and innovations to make the protocol more efficient, faster, and flexible for builders, without reducing any security guarantees.
+
+## Connections and Channels
+
+Instead of a legacy single-chain-single-connection protocol, Union uses a connection and channel standard to facilitate communication between rollups and chains. This has several advantages.
```
Comment (benluelo) [inline docs/src/content/docs/protocol/overview.mdx:13]: what standard are you referring to here as legacy

Comment (KaiserKarel) [reply docs/src/content/docs/protocol/overview.mdx:13]: Older tech like our competitors

Comment (benluelo) [reply docs/src/content/docs/protocol/overview.mdx:13]: ah so it's ibc vs <other>, not union vs ibc 👍

**Review by benluelo — APPROVED** [review]

**Review by aeryz — APPROVED** [review]

**Review by cor — APPROVED** [review]

**Pull request merged**